test/jdk/com/sun/crypto/provider/Cipher/ChaCha20/unittest/ChaCha20CipherUnitTest.java

```diff
@@ -66,32 +73,45 @@ public static void main(String[] args) throws Exception {
     private static void testTransformations() throws Exception {
         System.out.println("== transformations ==");
 
-        checkTransformation("ChaCha20", true);
-        checkTransformation("ChaCha20/None/NoPadding", true);
-        checkTransformation("ChaCha20-Poly1305", true);
-        checkTransformation("ChaCha20-Poly1305/None/NoPadding", true);
+        checkTransformation("ChaCha20", Expected.PASS);
+        checkTransformation("ChaCha20/None/NoPadding", Expected.PASS);
+        checkTransformation("ChaCha20-Poly1305", Expected.PASS);
+        checkTransformation("ChaCha20-Poly1305/None/NoPadding", Expected.PASS);
+
+        checkTransformation("ChaCha20-Poly1305/BAD_MODE/NoPadding", Expected.NSAE);
+        checkTransformation("ChaCha20-Poly1305/BAD_MODE/PKCS5Padding", Expected.NSAE);
 
-        checkTransformation("ChaCha20/ECB/NoPadding", false);
-        checkTransformation("ChaCha20/None/PKCS5Padding", false);
-        checkTransformation("ChaCha20-Poly1305/ECB/NoPadding", false);
-        checkTransformation("ChaCha20-Poly1305/None/PKCS5Padding", false);
+        checkTransformation("ChaCha20/ECB/NoPadding", Expected.NSAE_OR_NSPE);
+        checkTransformation("ChaCha20/None/PKCS5Padding", Expected.NSAE_OR_NSPE);
```

Comment:
Isn't this always going to throw NSPE given the current provider configuration? Also, same comment on line 86.